content/guides/end-to-end-testing/okta-authentication.md

```diff
@@ -70,11 +76,116 @@ require('dotenv').config()
 
 ## Custom Command for Okta Authentication
 
+There are two ways you can authenticate to Okta:
+
+- [Login with cy.origin()](/guides/end-to-end-testing/okta-authentication#Login-with-cy-origin)
+- [Programmatic Access](/guides/end-to-end-testing/okta-authentication#Programmatic-Login)
+
+### Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a custom command called `loginByOkta` to perform a login to
+[Okta](https://okta.com). This command will use
+[`cy.origin()`](/api/commands/origin) to
+
+1. navigate to the Okta origin
+2. input user credentials
+3. sign in and redirect back to the
+   [Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app)
+4. cache the results with [`cy.session()`](/api/commands/session)
+
+```jsx
+// cypress/support/auth-provider-commands/okta.ts
+// Okta
+const loginToOkta = (username: string, password: string) => {
+  Cypress.log({
+    displayName: 'OKTA LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    autoEnd: false,
+  })
+
+  cy.visit('/')
+  cy.origin(
+    Cypress.env('okta_domain'),
+    { args: { username, password } },
+    ({ username, password }) => {
+      cy.get('input[name="identifier"]').type(username)
+      cy.get('input[name="credentials.passcode"]').type(password, {
+        log: false,
+      })
+      cy.get('[type="submit"]').click()
+    }
+  )
+
+  cy.get('[data-test="sidenav-username"]').should('contain', username)
+}
+// right now our custom command is light. More on this later!
+Cypress.Commands.add('loginByOkta', (username: string, password: string) => {
+  return loginToOkta(username, password)
+})
+```
+
+Secondly, we can use our `loginByOkta` command in the test. Below is our test to
+login as a user via [Okta](https://okta.com) and run a few basic sanity tests.
+
+<Alert type="success">
+
+The
+[runnable version of this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui-auth-providers/okta.spec.ts)
```

Comment:
Is the custom command found in this file or the support file?

Reply:
support file, which in the code snippet at the top has something like `cypress/support/auth-provider-commands/okta.ts`. I'm hopiong thats enough to make it discoverable since that is what we've done elsewhere, even though just clicking on the custom command just takes you the the typings definition